### PR TITLE
Fix: Store Fee Quote for Payment

### PR DESCRIPTION
**Pull request by Tigh-Gherr**
- Add a `fee_rates` table to the database.
- Upsert any fee quote read during payment request from mapi into `fee_rates`
- Read the fee quote from `fee_rates` on receiving payment
- If expired, reject
- Also, remove destinations endpoint as this is just duplication

### CODE REVIEW

File: cmd/internal/dependencies.go [hunk 1]
```diff
@@ -61,16 +61,16 @@ func SetupRestDeps(cfg *config.Config, l log.Logger, db *sqlx.DB, c *client.Clie
 
 	seedSvc := service.NewSeedService()
 	privKeySvc := service.NewPrivateKeys(sqlLiteStore, cfg.Wallet.Network == "mainnet")
-	destSvc := service.NewDestinationsService(cfg.Wallet, privKeySvc, sqlLiteStore, sqlLiteStore, sqlLiteStore, mapiStore, seedSvc)
-	paymentSvc := service.NewPayments(l, spvv, sqlLiteStore, sqlLiteStore, sqlLiteStore, &paydSQL.Transacter{}, mapiStore, mapiStore, sqlLiteStore)
+	destSvc := service.NewDestinationsService(cfg.Wallet, privKeySvc, sqlLiteStore, sqlLiteStore, sqlLiteStore, seedSvc)
+	paymentSvc := service.NewPayments(l, spvv, sqlLiteStore, sqlLiteStore, sqlLiteStore, &paydSQL.Transacter{}, mapiStore, sqlLiteStore, sqlLiteStore)
```
Comment (jadwahab) [inline cmd/internal/dependencies.go:65]: that's sqlLiteStore twice, is that right?

Comment (Tigh-Gherr) [reply cmd/internal/dependencies.go:65]: Yep, `sqlLiteStore` implements a load of standalone func to fit pretty much every reader interface we have, so it is acting as a `DestinationReader`, `FeeReader`, `OwnerReader`, and `InvoiceReader` here.

File: data/sqlite/fees.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	VALUES (:invoice_id, :fee_json, :expires_at)
+	ON CONFLICT(invoice_id) DO UPDATE SET fee_json=:fee_json, expires_at=:expires_at
+`
+
+const sqlSelectFees = `
+	SELECT fee_json, expires_at FROM fee_rates
+	WHERE invoice_id = :invoice_id
+`
+
+func (s *sqliteStore) FeesQuoteCreate(ctx context.Context, args *payd.FeeQuoteCreateArgs) error {
```
Comment (jadwahab) [inline data/sqlite/fees.go:26]: typo: `FeeQuoteCreate`*

File: service/payments.go [hunk 3]
```diff
@@ -59,10 +59,13 @@ func (p *payments) PaymentCreate(ctx context.Context, args payd.PaymentCreateArg
 	if inv.State != payd.StateInvoicePending {
 		return lathos.NewErrDuplicate("D001", fmt.Sprintf("payment already received for invoice ID '%s'", args.InvoiceID))
 	}
-	fq, err := p.feeRdr.Fees(ctx)
+	fq, err := p.feeRdr.Fees(ctx, args.InvoiceID)
 	if err != nil {
 		return errors.Wrapf(err, "failed to read fees for payment with id %s", args.InvoiceID)
 	}
+	if fq.Expired() {
+		return lathos.NewErrUnprocessable("E001", "fee quote has expired, please make a new payment request")
```
Comment (jadwahab) [inline service/payments.go:67]: good shit

**Review by jadwahab — APPROVED** [review]

**Pull request merged**